### PR TITLE
Enhancement: Adds Webhook handling

### PR DESCRIPTION
**Pull request by silasjoisten**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,8 +56,106 @@ If `symfony/flex` is not available, or you prefer manual setup, follow these ste
...
+
+#### Configuration
+
+To enable Webhooks, add the following route to your application:
+
+```yaml
+# config/routes/storyblok.yaml
+storyblok:
+    resource: '@StoryblokBundle/config/routes.php'
+```
```
Comment (OskarStark) [inline README.md:80]: we can do this via recipe, right?

File: README.md [hunk 2]
```diff
@@ -56,8 +56,106 @@ If `symfony/flex` is not available, or you prefer manual setup, follow these ste
...
+
+```dotenv
+STORYBLOK_WEBHOOK_SECRET=your_webhook_secret
+```
+
+Once enabled, the bundle will automatically validate each Webhook request against this secret.
+
+#### Handling Webhook Events
+
+To process Webhooks, simply implement the `WebhookHandlerInterface`. The bundle automatically registers any classes
```
Comment (OskarStark) [inline README.md:107]:
```suggestion
To process Webhooks, implement the `WebhookHandlerInterface`. The bundle automatically registers any classes
```

File: README.md [hunk 3]
```diff
@@ -56,8 +56,106 @@ If `symfony/flex` is not available, or you prefer manual setup, follow these ste
...
+implementing this interface as Webhook handlers, so no additional service configuration is required.
```
Comment (OskarStark) [inline README.md:108]:
```suggestion
implementing this interface as Webhook handlers, no additional service configuration is required.
```

File: README.md (hunk 1 above)
Comment (silasjoisten) [reply README.md:80]: It will be yea but it will be commented out in the recipe.

File: src/Webhook/Event.php [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+ */
+
+namespace SensioLabs\Storyblok\Bundle\Webhook;
+
+use OskarStark\Enum\Trait\Comparable;
+
+/**
+ * @see https://www.storyblok.com/docs/guide/in-depth/webhooks
+ */
+enum Event: string
```
Comment (OskarStark) [inline src/Webhook/Event.php:21]: are these all available ones?

Comment (silasjoisten) [reply src/Webhook/Event.php:21]: yes.

File: src/Webhook/WebhookEventHandlerChain.php [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        if ([] === $handlers) {
+            throw new UnsupportedEventException(\sprintf('Event "%s" is not supported.', $event->value));
+        }
+
+        foreach ($handlers as $handler) {
+            $this->logger->info(\sprintf(
+                'Event %s handled by %s',
+                $event->value,
+                $handler::class,
+            ));
```
Comment (OskarStark) [inline src/Webhook/WebhookEventHandlerChain.php:63]: one line imho

**Review by OskarStark — APPROVED** [review]

File: tests/Util/TestHandler.php [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+/**
+ * This file is part of sensiolabs-de/storyblok-bundle.
+ *
+ * (c) SensioLabs Deutschland <info@sensiolabs.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace SensioLabs\Storyblok\Bundle\Tests\Util;
```
Comment (OskarStark) [inline tests/Util/TestHandler.php:14]: Should be under:
```suggestion
namespace SensioLabs\Storyblok\Bundle\Tests\Double;
```
namespace

I personally would name it "ConfigurableHandler", as you configure it via actor.

Comment (silasjoisten) [reply tests/Util/TestHandler.php]: How would you name the one which throws an exception then?

**Pull request merged**